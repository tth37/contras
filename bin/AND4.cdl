import And from  "And.cdl"
@Head
NAME And4
INPUT a[3:0]
OUTPUT out
SYMBOL And And1, And2, And3
@Body
And1(a = a[0], b = a[1]) => (out = a01)
And2(a = a[2], b = a[3]) => (out = a23) 
And3(a = a01, b = a23) => (out = a0123)
@End
OUTPUT out = a0123