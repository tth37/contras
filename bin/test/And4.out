@Case
  a[0] = 0101010101010101
  a[1] = 0011001100110011
  a[2] = 0000111100001111
  a[3] = 0000000011111111
out[0] = 0000000000000001
